src/linter/ui5Types/fix/EventHandlersFix.ts

```diff
@@ -0,0 +1,124 @@
+import ts, {Program} from "typescript";
+import {PositionInfo} from "../../LinterContext.js";
+import {Attribute, Position, SaxEventType} from "sax-wasm";
+import XmlEnabledFix from "./XmlEnabledFix.js";
+import {type AttributeDeclaration} from "../../xmlTemplate/xmlNodes.js";
+import {ChangeAction, ChangeSet} from "../../../utils/textChanges.js";
+import SourceFileMetadataCollector from "../SourceFileMetadataCollector.js";
+
+export default class EventHandlersFix extends XmlEnabledFix {
+	protected sourcePosition: PositionInfo | undefined;
+	protected startPos: number | undefined;
+	protected endPos: number | undefined;
+	protected trailingCommaPos: number | undefined;
+	private fullMethodSignature: string | undefined;
+	private hasAvailableController = false;
+
+	constructor(
+		private methodName: string,
+		private controllerName?: string
+	) {
+		super();
+	}
+
+	visitLinterNode(_node: ts.Node | AttributeDeclaration, sourcePosition: PositionInfo) {
+		this.sourcePosition = sourcePosition;
+		// If controller name is not present we cannot determine which is the actual controller, so skip further checks
+		return !!this.controllerName;
+	}
+
+	getNodeSearchParameters() {
+		if (this.sourcePosition === undefined) {
+			throw new Error("Position for search is not defined");
+		}
+		return {
+			nodeTypes: [ts.SyntaxKind.PropertyAssignment],
+			xmlEventTypes: [SaxEventType.Attribute],
+			position: this.sourcePosition,
+		};
+	}
+
+	methodExistsInController(
+		tsProgram?: Program,
+		checker?: ts.TypeChecker,
+		metadataCollector?: SourceFileMetadataCollector
```

Comment:
Why are those optional? Currently I don't see cases where we don't provide them. And we currently need all of them.

Reply:
Done